toolchain/check/handle_name.cpp

```diff
@@ -22,6 +23,14 @@ auto HandleParseNode(Context& context, Parse::MemberAccessExprId node_id)
     auto member_id =
         PerformCompoundMemberAccess(context, node_id, base_id, member_expr_id);
     context.node_stack().Push(node_id, member_id);
+  } else if (context.node_stack().PeekIs<Parse::NodeKind::IntLiteral>()) {
```

Comment:
Here and below, I think the repeated `PeekIs` (for both ParenExpr and IntLiteral) is going to lead to the node kind being queried. To avoid the duplicate logic, can you store the valid of the kind and compare? i.e., something like:

```
auto node_kind = context.node_stack().PeekNodeKind();
if (node_kind == Parse::NodeKind::ParenExpr) {
  ...
} else if (node_kind == Parse::NodeKind::IntLiteral) {
```